### PR TITLE
initial setup to handle previews

### PR DESCRIPTION
**Pull request by saito-sv**

### CODE REVIEW

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-task/saito/previews.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-task/saito/previews.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-task-saito-previews.dev-app.axial.exchange

File: infra/dev/cloudfront.tf [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
+    }
   }
 
-  # fixes for SPA
-  custom_error_response {
-    error_code         = 400
-    response_code      = 200
-    response_page_path = "/index.html"
-  }
+  # # fixes for SPA
```
Comment (saito-sv) [inline infra/dev/cloudfront.tf]: need to remove this

File: infra/dev/lambdas/preview-origin/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  console.log("request made")
+    const { request } = event.Records[0].cf;
+    try {
+      const host = request.headers['x-forwarded-host'][0].value;
+      const branch = host.match(/^preview-([^\.]+)/)[1];
+      request.origin.custom.path = `/${branch}`;
+    } catch (e) {
+     callback(null, request)
+    }
+    console.log(request)
```
Comment (saito-sv) [inline infra/dev/lambdas/preview-origin/index.js]: remove console log debug

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-task-saito-previews.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-task-saito-previews.dev-app.axial.exchange

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-task-saito-previews.dev-app.axial.exchange

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**